lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -252,7 +254,11 @@ def add_payment_method(xml, amount, payment_method, options)
           end
         else
           xml.tag! 'paymentDetails', credit_fund_transfer_attribute(options) do
-            xml.tag! CARD_CODES[card_brand(payment_method)] do
+            card_code = CARD_CODES[card_brand(payment_method)]
+            if card_code.nil?
+              card_code = UNKNOWN_CARD_CODE
+            end
+            xml.tag! card_code do
```

Comment:
😎 The logic checks out here but I'm wondering if it makes sense to abstract into a private method. Something like: `xml.tag! card_code_for(payment_method)`.

Reply:
Certainly - that would be a cleaner way for sure.